### PR TITLE
Refactored hubs and merged hub search with menu

### PR DESCRIPTION
**Pull request by Geefy**

### CODE REVIEW

File: source/webclient/XPower_WebClient/src/app/Services/hub.service.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  constructor() {
+    
+  }
+
+  /*
+  Retrieves the socket devices from the hub
+  */
+  public GetHubDevicesFromHome(home_idenfitifer: string) : void {
+    //An offline hub for testing
+    let a = new HubDevice();
```
Comment (Ripiter) [inline source/webclient/XPower_WebClient/src/app/Services/hub.service.ts:24]: better name

File: source/webclient/XPower_WebClient/src/app/Components/hub/hub.component.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  constructor() {
+  }
+
+  ngOnInit(): void {
+  }
+
+  /*
+  Event for when card is double clicked
+  */
+  ondblclickevent(event : any): void {
```
Comment (Ripiter) [inline source/webclient/XPower_WebClient/src/app/Components/hub/hub.component.ts:32]: camel case

File: source/webclient/XPower_WebClient/src/app/Pages/hub-searcher/hub-searcher.component.scss [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    top: 50%;
+    left: 50%;
+    -ms-transform: translate(-50%, -50%);
+    transform: translate(-50%, -50%);
+    
+}
+mat-form-field{
+    border-radius: 5px;
+    margin-top: 10%;
+    color:rgba(27,40,69,1);
```
Comment (Ripiter) [inline source/webclient/XPower_WebClient/src/app/Pages/hub-searcher/hub-searcher.component.scss:13]: create global css for button 
with color
and with backgroud

**Review by Ripiter — CHANGES_REQUESTED** [review]
few comments

**Comment by blackout1471** [comment]
Where is unit tests?

File: source/webclient/XPower_WebClient/src/app/Converters/Json/SocketDeviceJsonRequestConverter.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { SocketDeviceRequestConverter } from "../SocketDeviceRequestConverter";
+
+export class SocketDeviceJsonRequestConverter implements SocketDeviceRequestConverter {
+
```
Comment (blackout1471) [inline source/webclient/XPower_WebClient/src/app/Converters/Json/SocketDeviceJsonRequestConverter.ts:4]: Needs comments

**Review by blackout1471 — CHANGES_REQUESTED** [review]

**Review by blackout1471 — APPROVED** [review]

**Review by Ripiter — APPROVED** [review]
looks better

**Pull request merged**